### ISSUE TITLE
pin-project 0.3.3 introduces breaking changes about "missing extern crate core" compared to 0.3.2

### ISSUE THREAD

**Issue by shepmaster** [taiki-e/pin-project#16]
### 0.3.2

```
$ cargo update -p pin-project --precise 0.3.2
    Updating crates.io index
    Updating pin-project v0.3.3 -> v0.3.2

$ cargo test
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running target/debug/deps/futures_compat-27fb39e639db3ee7

running 1 test
test implements_error ... ok
```

### 0.3.3

```
$ cargo update -p pin-project --precise 0.3.3
    Updating crates.io index
    Updating pin-project v0.3.2 -> v0.3.3

$ cargo test
   Compiling snafu v0.4.1 (/Users/shep/Projects/snafu)
error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_future.rs:127:18
    |
127 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_future.rs:166:18
    |
166 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_stream.rs:129:18
    |
129 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error[E0433]: failed to resolve: maybe a missing `extern crate core;`?
   --> /Users/shep/Projects/snafu/src/futures/try_stream.rs:171:18
    |
171 | #[unsafe_project(Unpin)]
    |                  ^^^^^ maybe a missing `extern crate core;`?

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0433`.
error: Could not compile `snafu`.
```

**Comment by taiki-e** [comment]
@shepmaster Thanks for the reporting! 
I yanked 0.3.3.

I think that is related to 7efdfc61d49a9cfa530de7b2ec29a55081f9b416 or eee7fa9c6143e80256c2b5a284c55f20dad9e142. I will investigate.

**Comment by taiki-e** [comment]
The cause is this code added in 7efdfc61d49a9cfa530de7b2ec29a55081f9b416:
https://github.com/taiki-e/pin-project/blob/7575fb99ffd0bce1da5219a7e42cc8626f3fe5b6/src/unsafe_project/mod.rs#L65-L67

`Span` contains information about the edition. So if pin-project is used in 2015 edition's crate, `::core::marker::Unpin` is handled as code defined in the 2015 edition. Then 2015 edition's import rule is applied.

**Comment by taiki-e** [comment]
Published 0.3.4 which fixes this issue.
